cmd/flags.go

```diff
@@ -239,6 +240,23 @@ func readConfig(ctx *cli.Context) (*v1beta1.Cluster, error) {
 			cfg.Spec.K0s.Config.Merge(k0s)
 		}
 	}
+	otherConfigs := mr.FilterResources(func(rd *manifest.ResourceDefinition) bool {
+		if strings.EqualFold(rd.APIVersion, v1beta1.APIVersion) && strings.EqualFold(rd.Kind, "cluster") {
+			return false
+		}
+		if strings.EqualFold(rd.APIVersion, "k0s.k0sproject.io/v1beta1") && strings.EqualFold(rd.Kind, "clusterconfig") {
+			return false
+		}
+		return true
+	})
+	if len(otherConfigs) > 0 {
+		cfg.Metadata.Manifests = make(map[string][]byte)
+		log.Debugf("found %d additional resources in the configuration", len(otherConfigs))
+		for _, otherConfig := range otherConfigs {
+			log.Debugf("found resource: %s (%d bytes)", otherConfig.Filename(), len(otherConfig.Raw))
```

Comment:
Aren't we loading all `otherConfigs` from a single yaml doc? In this case, what does `otherConfig.Filename()` refer to?

Reply:
k0sctl accepts multiple `--config`s or `--config configs/` or `--config "configs/k0s*.yaml"`, not only a single multidoc yaml. The manifest reader tries to keep track of their origins for messages, but if it came from stdin or it doesn't know the filename for some other reason, it either just calls it "manifest" (while preliminary reading/unmarshaling fails) or calls it `fmt.Sprintf("%s-%s-%d.yaml", safeFn(rd.APIVersion), safeFn(rd.Kind), time.Now().UnixNano())`.